pkg/engine/background.go

```diff
@@ -96,10 +96,18 @@ func (e *engine) filterRule(
 	excludeGroupRole := e.configuration.GetExcludedGroups()
 	namespaceLabels := policyContext.NamespaceLabels()
 
-	if err := MatchesResourceDescription(subresourceGVKToAPIResource, newResource, rule, admissionInfo, excludeGroupRole, namespaceLabels, "", policyContext.SubResource()); err != nil {
+	mock := false
+	if policyContext.JSONContext() != nil {
+		resource, err := policyContext.JSONContext().Query("isMock")
```

Comment:
IMO it's not a good idea to manipulate the json context.

Reply:
We need to pass the flag down to the required method. `contextLoader.go` also adds to the json context.
https://github.com/kyverno/kyverno/blob/2f6450546901cecff9022e4e8cd18814047f2687/pkg/engine/api/contextloader.go#L44-L55

Do you want me to use:
```go
AddContextEntry(name string, dataRaw []byte) error
```
How do you propose to send the flag from CLI to validation? I don't want to modify the `validate()` function signature to include to flag.

Reply:
@eddycharly ping, let's get this also done, can you share the approach in your mind?

Reply:
@eddycharly - can you help here?